### PR TITLE
Bug/2221 invalid error updating non existing attribute

### PR DESCRIPTION
**Pull request by kzangeli**
Fixed issue #2221 
Now, the issue is describing an older behavior of the broker, part of the issue has been fixed by another PR. In this PR, the request in #2221 is made receive a 'Attribute Not Found' instead of the former 'Entity Not Found'  (the entity exists but doesn't have the attribute that issue #2221 tries to replace).

Also, plenty of older 'context element not found' texts replaced for the nicer message 'The requested entity has not been found. Check type and id', and 'Attribute Not Found' is also used in more places than before.

### CODE REVIEW

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -23,3 +23,4 @@
 - Fix: invalid regex patterns detection (#968)
 - Fix: Better error returned on invalid geoquery (Issue #2174)
 - Fix: Better error returned on attribute not found in GET /v2/entities/<entity-id>/attrs/<attr-name>/value (Issue #2220)
+- Fix: Correct error returned when trying to replace a non-existing attribute (Issue #2221)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: Maybe the operation that has been fixed (PUT /v2/entities/id/attrs/attrName) should be mentioned?

File: src/lib/mongoBackend/MongoCommonUpdate.cpp [hunk 2]
```diff
@@ -3053,14 +3053,22 @@ void processContextElement
       if (forwardsPending(responseP) == false)
       {
         cerP->statusCode.fill(SccContextElementNotFound);
-        responseP->oe.fill(SccContextElementNotFound, "No context element found", "NotFound");
+
+        if (apiVersion == "v1")
+        {
+          responseP->oe.fill(SccContextElementNotFound, "Entity not found. Check type and id", "NotFound");
```
Comment (fgalan) [inline src/lib/mongoBackend/MongoCommonUpdate.cpp]: nit: third parameter is not needed for v1 (althout it doesn't hurt, anyway...)

File: src/lib/serviceRoutines/postUpdateContext.cpp [hunk 3]
```diff
@@ -690,7 +690,14 @@ std::string postUpdateContext
     if (fails == response.contextElementResponseVector.size())
     {
       // If all CER result in error, then it isn't a partial update, but a regular NotFound
-      parseDataP->upcrs.res.oe.fill(SccContextElementNotFound, "No context element found", "NotFound");
+      if (ciP->apiVersion == "v1")
+      {
+        parseDataP->upcrs.res.oe.fill(SccContextElementNotFound, "No context element found", "NotFound");
```
Comment (fgalan) [inline src/lib/serviceRoutines/postUpdateContext.cpp]: nit: third parameter is not needed for v1 (althought it doesn't hurt, anyway...)

File: CHANGES_NEXT_RELEASE (hunk 1 above)
Comment (kzangeli) [inline CHANGES_NEXT_RELEASE]: Fixed in 248de8e

File: src/lib/mongoBackend/MongoCommonUpdate.cpp (hunk 2 above)
Comment (kzangeli) [inline src/lib/mongoBackend/MongoCommonUpdate.cpp]: ok, but with it the calls looks similar between v1 and v2.

However, I just noticed that I have changed the text also for v1 here ... That's not what I intented to do ...

Rollback of that part?

Comment (fgalan) [inline src/lib/mongoBackend/MongoCommonUpdate.cpp]: Discussed at skype. Rollback.

Comment (kzangeli) [inline src/lib/mongoBackend/MongoCommonUpdate.cpp]: Fixed in 4066516

File: src/lib/serviceRoutines/postUpdateContext.cpp (hunk 3 above)
Comment (kzangeli) [inline src/lib/serviceRoutines/postUpdateContext.cpp]: NTC

**Comment by fgalan** [comment]
LGTM

**Pull request merged**